### DISCUSSION TITLE
Live Streaming on Youtube with OBS and Live Translation to English with AURA

### DISCUSSION THREAD

**Discussion by micoach-agustin** [deepgram/community#736] [General help]
Dear friends, I am creating a number of LIVE Events in Spanish on Youtube by OBS. So, I want to clone the event and use Deepgram API to do the translation in English with AURA (with a delay of 30 seconds or 60 seconds) so I can have both Live Events simultaneously. Any ideas on how to program it with python?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment]
Can you describe a bit more what your setup looks like? We don't offer a translation (Spanish -> English) service that would work well for this use-case I think, but here is a demo that can transcribe Youtube Livestreams: https://github.com/nikolawhallon/youtube-deepgram-livestream

**Reply by micoach-agustin** [reply]
Thanks Niko, i will check it tonight :)

@team-deepgram What I want to accomplish
> Step 1. Transcribe with Deepgram a LIVE youtube event (this format: https://youtube.com/live/XXXXXXXX?feature=share but that it's not online yet, I am producing it with OBS on my computer, so we have a latency of 25 seconds aprox. **First question:** How to do live transcription with Deepgram on local computer at the studio. 
> Step 2. Translate it with Google API (I think can be a good solution) to English.
> Step 3. Run Aura to do a English text-to-speech 
> Step 4. Use this audio as the output audio to a new channel LIVE on Youtube (different from the previous one) so people can access to https://youtube.com/live/YYYYYYYY?feature=share and see it on English and sincronize it with the original video

**Reply by nikolawhallon** [reply]
Step 1: can be accomplished with the demo script above

Step 1.5: if you don't want to grab the audio from the youtube livestream, but from your computer's mic itself live, there are docs/tutorials for that too: https://developers.deepgram.com/docs/getting-started-with-the-streaming-test-suite#stream-your-microphone

Step 2: could add some logic to that python client which pieces together incoming transcripts and sends off to Google's API for translation

Step 3: in the same script, take the results from Google's API and make requests to DG Aura

Step 4: this is where my knowledge fails me! I wonder if there is either (A) a programmatic way to pipe audio into OBS as an audio source (there must be, right??) or (B) some youtube API for injecting audio in a livestream?
